### PR TITLE
Add 'ACK' to PR Dashboard, to hide items in 'Needs Attention'.

### PR DESCRIPTION
**Pull request by rmmh**
Related issue: #697.

### CODE REVIEW

File: gubernator/view_pr.py [hunk 1]
```diff
@@ -96,6 +97,20 @@ def get(self, path, pr):
             max_builds=max_builds, header=headings, repo=repo, rows=rows, path=path))
 
 
+def get_acks(login, prs):
```
Comment (fejta) [inline gubernator/view_pr.py]: Description?

File: gubernator/view_pr.py [hunk 2]
```diff
@@ -96,6 +97,20 @@ def get(self, path, pr):
...
+    acks = {}
+    result = ghm.GHUserState.make_key(login).get()
+    if result:
+        acks = result.acks
+        if prs:
+            stale = set(acks) - set(pr.key.id() for pr in prs)
```
Comment (fejta) [inline gubernator/view_pr.py]: It is not clear to me what prs means... prs in what kind of state? I guess PRs still needing my attention?

File: gubernator/templates/pr_dashboard.html [hunk 3]
```diff
@@ -40,10 +45,21 @@ <h3 class="pr-section">
 			<td>{{pl['assignees']|join(", ")}}
 			<td>{{pl|classify_size}}
 			<td>{{pl['title']}}
+			% if title == "Needs Attention" and acks is not none
```
Comment (fejta) [inline gubernator/templates/pr_dashboard.html]: nit: Did you try putting it on the left?

File: gubernator/main.py [hunk 4]
```diff
@@ -20,7 +20,7 @@
 import webapp2
 from webapp2_extras import security
 
-from google.appengine.api import app_identity
+from google.appengine.api import app_identity, modules
```
Comment (fejta) [inline gubernator/main.py]: One line per import, please

https://google.github.io/styleguide/pyguide.html?showone=Packages#Packages

File: gubernator/github/admin.py [hunk 5]
```diff
@@ -47,7 +47,7 @@ def handle_entity(entity):
 @ndb.toplevel
 def migrate(migration, cursor=None, last_parent=None, stop=False):
     entities, next_cursor, more = migration.query().fetch_page(
-        10, start_cursor=cursor, keys_only=migration.keys_only)
+        1, start_cursor=cursor, keys_only=migration.keys_only)
```
Comment (fejta) [inline gubernator/github/admin.py]: Add a note explaining why?

**Comment by fejta** [comment]
Looks like you're pulling in a bad set of commits?

File: gubernator/templates/pr_dashboard.html [hunk 6]
```diff
@@ -40,10 +45,21 @@ <h3 class="pr-section">
...
+			% endif
 		% endfor
 	</table>
 	% else
 		<p><h4 class="pr-section">No Results</h4>
 	% endif
 % endfor
+
+<div id="help">
+<p><b>Needs Attention</b> is a heuristic grouping based on a <a href="https://github.com/kubernetes/test-infra/blob/master/gubernator/github/classifier.py#L288">simple state machine</a>. Use the "ACK" button to to remove a PR from "Needs Attention" until something new occurs.
```
Comment (apelisse) [inline gubernator/templates/pr_dashboard.html]: > Use the "ACK" button to to remove a PR from "Needs Attention"

image

Dear Ryan, there are too many "to". Please remove one.

**Pull request merged**